src/main/scala/tech/cryptonomic/conseil/tezos/michelson/parser/JsonParser.scala

```diff
@@ -38,6 +39,17 @@ object JsonParser {
     def toMichelsonExpression: MichelsonExpression
   }
 
+  type EmbeddedElement = Either[Either[JsonExpression, JsonInstruction], List[JsonInstruction]]
```

Comment:
Minor: Is there a way to use an ADT for this instead of "nested disjunctions"? Personally I find them hard to parse mentally and to reason with.